docs/quickstart/junit_4_quickstart.md

```diff
@@ -0,0 +1,77 @@
+# JUnit 4 Quickstart
+
+It's easy to add Testcontainers to your project - let's walk through a quick example to see how.
+
+Let's imagine we have a simple program that has a dependency on Redis, and we want to add some tests for it.
+ 
+You can see the implementation code and the corresponding test before we've added Testcontainers below:
+
+!!! example "Scenario code"
+    <!--codeinclude-->
+    [Implementation](../examples/src/main/java/quickstart/RedisBackedCache.java) block:RedisBackedCache
+    [Pre-Testcontainers test code](../examples/src/test/java/quickstart/RedisBackedCacheIntTestStep0.java) block:RedisBackedCacheIntTestStep0
+    <!--/codeinclude-->
+
+Notice that the existing test has a problem - it's relying on a local installation of Redis, which is a red flag for test reliability.
+This may work if we were sure that every developer and CI machine had Redis installed, but would fail otherwise.
+We might also have problems if we attempted to run tests in parallel, such as state bleeding between tests, or port clashes.
+
+Let's start from here, and see how to improve the test with Testcontainers:  
+
+## 1. Add Testcontainers as a test-scoped dependency
+
+First, add Testcontainers as a dependency as follows:
+
+```xml tab='Maven'
+<dependency>
+    <groupId>org.testcontainers</groupId>
+    <artifactId>testcontainers</artifactId>
+    <version>{{latest_version}}</version>
+    <scope>test</scope>
+</dependency>
+```
+
+```groovy tab='Gradle'
+testRuntime "org.testcontainers:testcontainers:{{latest_version}}"
+```
+
+## 2. Get Testcontainers to run a Redis container during our tests
+
+Simply add the following to the body of our test class:
+
+<!--codeinclude-->
+[JUnit 4 Rule](../examples/src/test/java/quickstart/RedisBackedCacheIntTest.java) inside_block:rule
+<!--/codeinclude-->
+
+The `@Rule` annotation tells JUnit to notify this field about various events in the test lifecycle.
+In this case, our rule object is a Testcontainers `GenericContainer`, configured to use a recent Redis image from Docker Hub, and configured to expose a port.
+
+If we run our test as-is, then regardless of the actual test outcome, we'll see logs showing us that Testcontainers:
+
+* was activated before our test method ran
+* discovered and quickly tested our local Docker setup
+* pulled the image if necessary
+* started a new container and waited for it to be ready
+* shut down and deleted the container after the test
+
+## 3. Make sure our code can talk to the container
+
+Before Testcontainers, we might have hardcoded an address like `localhost:6379` into our tests.
+
+Testcontainers uses *randomized ports* for each container it starts, but makes it easy to obtain the actual port at runtime.
+We can do this in our test `setUp` method, to set up our component under test:
+
+<!--codeinclude-->
+[Obtaining a mapped port](../examples/src/test/java/quickstart/RedisBackedCacheIntTest.java) inside_block:setUp
+<!--/codeinclude-->
```

Comment:
I would also add a warning block, something like:
> do *not* hardcode `localhost`. While it might work for some environments, it will break on others

Reply:
Good idea - adding.